store/tikv/region_cache.go

```diff
@@ -171,11 +171,31 @@ func (r *RegionStore) kvPeer(seed uint32) AccessIndex {
 		}
 		candidates = append(candidates, AccessIndex(i))
 	}
-
+	candidates = r.filterStoreCandidates(op, candidates...)
 	if len(candidates) == 0 {
 		return r.workTiKVIdx
 	}
-	return candidates[int32(seed)%int32(len(candidates))]
```

Comment:
uint32 convert to int32 would cause overflow to negative number, thus I fix it here.